packages/opal-server/opal_server/server.py

```diff
@@ -352,6 +356,15 @@ async def start_server_background_tasks(self):
                             "listening on webhook topic: '{topic}'",
                             topic=opal_server_config.POLICY_REPO_WEBHOOK_TOPIC,
                         )
+                          #the leader should be the only one to constantly push data config pushes to clients
+                        logger.info(
+                            "Binding data update publisher to: {pid}",
+                            pid=os.getpid(),
+                        )
+                        #bind data updater publishers to lead worker
+                        data_publisher = DataUpdatePublisher(self.publisher)
+                        for polling_update in data_publisher.create_polling_updates(opal_server_config.DATA_CONFIG_SOURCES):
```

Comment:
Class is updated.

As far as the cleanup goes, it seems because we're using https://fastapi-utils.davidmontague.xyz/user-guide/repeated-tasks/#the-repeat_every-decorator that it auto cleans up during the Shutdown event. At least that's what I guess is implied by:

>If you also apply the @app.event("startup") decorator, FastAPI will call the function during server startup, and the function will then be called repeatedly while the server is still running.